drop-in-checkout/src/main/java/io/primer/checkout/dropin/checkout/data/repository/PrimerDropInRepository.kt

```diff
@@ -0,0 +1,82 @@
+package io.primer.checkout.dropin.checkout.data.repository
+
+import android.content.Context
+import dagger.hilt.android.qualifiers.ApplicationContext
+import io.primer.android.Primer
+import io.primer.android.PrimerCheckoutListener
+import io.primer.android.completion.PrimerErrorDecisionHandler
+import io.primer.android.core.logging.PrimerLogLevel
+import io.primer.android.core.logging.PrimerLogging
+import io.primer.android.domain.PrimerCheckoutData
+import io.primer.android.domain.error.models.PrimerError
+import io.primer.checkout.dropin.BuildConfig
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import javax.inject.Inject
+
+interface PrimerDropInRepository {
+
+    fun start(clientToken: String)
+
+    fun setCustomErrorMessage(message: String?)
```

Comment:
I don't like I had to add this method here, but I needed a way to pass in the string to be displayed as custom error message when the payment fails. 
Even if I have access to application context here, I didn't like the idea of bringing in view specific dependencies into a repository. 
If you have a better idea of how I could do this: `errorHandler?.showErrorMessage(customErrorMessage)` please let me know.

Reply:
After some discussions with @semirp we decided it would be best to have an error mapper injected into the repository which will take care of preparing the error messages. Therefore the code in the previous comment was removed.